include/core/Layer.hpp

```diff
@@ -0,0 +1,147 @@
+#ifndef __NGEN_LAYER__
+#define __NGEN_LAYER__
+
+#include "LayerData.hpp"
+#include "Simulation_Time.hpp"
+
+#ifdef NGEN_MPI_ACTIVE
+#include "HY_Features_MPI.hpp"
+#else
+#include "HY_Features.hpp"
+#endif
+
+#include "Pdm03.h"
+
+namespace ngen
+{
+
+    class Layer
+    {    
+        public:
+
+        #ifdef NGEN_MPI_ACTIVE
+            typedef hy_features::HY_Features_MPI feature_type;
+        #else
+            typedef hy_features::HY_Features feature_type;
+        #endif
+
+        Layer(
+                const LayerDescription& desc, 
+                const std::vector<std::string>& p_u, 
+                const Simulation_Time& s_t, 
+                feature_type& f, 
+                geojson::GeoJSON cd, 
+                std::shared_ptr<pdm03_struct> pdmd,
+                long idx) :
+            description(desc),
+            processing_units(p_u),
+            simulation_time(s_t),
+            features(f),
+            catchment_data(cd),
+            pdm_et_data(pdmd),
+            output_time_index(idx)
+        {
+
+        }
+
+        virtual ~Layer() {}
+
+        /***
+         * @brief Return the next timestep that will be processed by this layer in epoc time units
+        */
+        time_t next_timestep_epoch_time() { return simulation_time.next_timestep_epoch_time(); }
+
+
+        /***
+         * @brief Return the last timesteo that was processed by this layer in epoc time units
+        */
+        time_t current_timestep_epoch_time() { return simulation_time.get_current_epoc_time(); }
```

Comment:
```suggestion
        time_t current_timestep_epoch_time() { return simulation_time.get_current_epoch_time(); }
```
Or is there really a typo in the function name? If so, please fix that too!

Reply:
You can ask Matt on that one he wrote the simulation time class (corrected the typo)